File: TwoPointer/SubarrayProductLessThanK.java
public class SubarrayProductLessThanK {
    public static void main(String[] args) {
        SubarrayProductLessThanK solution = new SubarrayProductLessThanK();
        int[] nums = { 10, 5, 2, 6 };
        int k = 100;
        int ans = solution.input(nums, k);
        System.out.println(ans);
    }

    public int input(int[] nums, int k) {
        int fast = 0, last = 1, j = 0;
        for (int i = 0; i < nums.length; i++) {
            last = last * nums[i];
            while (last >= k && j < i) {
                last /= nums[j];
                j++;
            }
            if (last < k) {
                fast += (i - j + 1);
            }
        }
        return fast;
    }
}

File: TwoPointer/SubarrayProductLessThanK.js
var SubarrayProductLessThanK=function(nums,k){
let left=0;
    let right =1;
    let j=0;
    for(let i=0;i<nums.length;i++){
        right=right*nums[i];
        if(right>=k&&j<i){
            right/=nums[j];
            j++;
        }
        if(right<k){
            left+=i-j+1;
        }
    }
    return left;
};
const nums=[10,5,2,6];
const k=100;
const result=SubarrayProductLessThanK(nums,k);
console.log(result);